MTA-6811, 6812, 6813, 6814, 6815, 6816 CQA April Developer Lightspeed work (#357)
migtools/mta-documentation: docs/topics/developer-lightspeed/proc_apply-rag-resolution.adoc

= Applying resolutions generated by the solution server

[role="_abstract"]

> When you request code resolutions by enabling the Solution Server, an issue displays the success metric when the metric becomes available. A success metric indicates the confidence level in applying the fix suggestion from the LLM based on how many times you applied the update in past analyses.

Reviewer: This is a procedure module. A short description must include why and what to comply with the requirements. We can move this information below the short description, if you'd like to preserve it:
```suggestion
To resolve application code issues, apply the fix suggestions generated by the Solution Server.

When you request code resolutions by enabling the Solution Server, issues display a success metric when it becomes available. This metric indicates the confidence level of the suggested LLM fix based on your past updates.
```

You can review the code updates and edit the suggested code resolutions before accepting the suggestions. 

.Prerequisites

* You opened a Java project in your Visual Studio Code workspace.
* You configured a profile on the *{ProductShortName} Analysis View* page.
* You ran an analysis after enabling the Solution Server.

.Procedure

. Review the issues from the *Analysis results* space of the *{ProductShortName} view analysis* page by the following tabs: 
.. *All*: lists all incidents identified in your project.
.. *Files*: lists all the files in your project for which the analysis identified issues that you must resolve. 
.. *Issues*: lists all issues across different files in your project.
. Use the *Category* drop down to filter issues based on how crucial the fix is for the target migration. You can filter mandatory, potential, and optional issues. 

> . Click *Has Success Rate* to check how many times you accepted the same issue resolution in an earlier analysis.

Reviewer: ```suggestion
. Click *Has Success Rate* to check how many times you accepted the same issue resolution in a previous analysis.
```

> . Click the solution tool to trigger automated updates to your code. If you applied any category filter, {mta-dl-plugin} makes code updates for all incidents, specific files, or specific issues based on the filter.

Reviewer: ```suggestion
. Click the solution tool to trigger automated updates to your code. If you applied any category filter, {mta-dl-plugin} applies code updates for all incidents, specific files, or specific issues based on the filter.
```

> . Click the solution tool to trigger automated updates to your code. If you applied any category filter, {mta-dl-plugin} applies the updates for all incidents, specific files, or specific issues based on the filter.

Reviewer: ```suggestion
. Click the solution tool to trigger automated updates to your code. If you applied any category filter, {mta-dl-plugin} applies the updates to all incidents, specific files, or specific issues based on the filter.
```
